scripts/gen_regex.py

```diff
@@ -4,89 +4,74 @@
 from pkg_resources import resource_filename
 
 
+CATEGORIES = [unicodedata.category(chr(i)) for i in range(0x110000)]
 DATA_PATH = pathlib.Path(resource_filename('wordfreq', 'data'))
 
 
+def func_to_regex(func):
+    """
+    Given a function that returns True or False for a numerical codepoint,
+    return a regex character class accepting the characters resulting in True.
+    Ranges separated only by unassigned characters are merged for efficiency.
```

Comment:
I agree with reducing assumptions we make about unassigned characters in general, but when it comes to emoji, the assumption we make in ftfy is that a codepoint in the emoji block that's not assigned is simply not assigned *yet*.

...But it looks like we're letting ftfy tell us what's an emoji anyway, so maybe that acts like the character is assigned for the purposes of this function. In that case, great.